### ISSUE TITLE
Can't use yarn in context of another command

### ISSUE THREAD

**Issue by JanPokorny** [pkgxdev/pkgx#994]
When using `pkgx +yarn -- command`, this happens:

```
❯ pkgx +yarn -- bash
× multiple projects provide: yarn
│ pls be more specific:
│ 
│     pkgx +classic.yarnpkg.com +yarn -- bash 
│     pkgx +yarnpkg.com +yarn -- bash 
│ 
╰─➤ https://docs.pkgx.sh/help/ambiguous-pkgspec

❯ pkgx +classic.yarnpkg.com +yarn -- bash
× multiple projects provide: yarn
│ pls be more specific:
│ 
│     pkgx +classic.yarnpkg.com +classic.yarnpkg.com +yarn -- bash 
│     pkgx +yarnpkg.com +classic.yarnpkg.com +yarn -- bash 
│ 
╰─➤ https://docs.pkgx.sh/help/ambiguous-pkgspec

❯ pkgx +yarnpkg.com +yarn -- bash
× multiple projects provide: yarn
│ pls be more specific:
│ 
│     pkgx +classic.yarnpkg.com +yarnpkg.com +yarn -- bash 
│     pkgx +yarnpkg.com +yarnpkg.com +yarn -- bash 
│ 
╰─➤ https://docs.pkgx.sh/help/ambiguous-pkgspec
```

Furthermore to the error text being clearly wrong, the URL 404s.

**Comment by jhheider** [comment]
`+yarn` is ambiguous. you either want: `pkgx +classic.yarnpkg.com -- bash` or `pkgx +yarnpkg.com -- bash`. the error message is _definitely_ misleading.

**Comment by JanPokorny** [comment]
@jhheider What's the purpose of having a separate `classic.yarnpkg.com`? As opposed to simply having `yarn@1`.

**Comment by jhheider** [comment]
We debated the correct way to handle it. Basically yarnv1 and berry are pretty different and both under active development. Tt was decided that breaking them up was the correct path forward. A similar choice was made, though for slightly different reasons, with pcre.org and pcre.org/v2. It's possible we should have done the same thing with openssl v1.1 and v3, and that has created some but not a lot of issues.

**Comment by JanPokorny** [comment]
@jhheider I see, thanks for clarifying. As a feedback from someone just starting with / evaluating pkgx, this is a bit confusing design. Other package managers either have:

- per-version packages for cases where this is appropriate (eg. `brew`, `apt`, `scoop`) which keep only the newest version packaged for each (eg. `python@3.9` on Brew is Python 3.9.19 and there is no easy way to go to eg. 3.9.18)
- full version support (eg. `asdf`, `pip`, `npm`) where a version range can be specified when installing a package

I initially thought that pkgx falls into the second category (since you can append `@version` to any package you wish, including minor versions), but the Yarn split really seems like the first category, so both approaches are combined here in a sort of confusing way. As a user, I would expect `pkgx yarn` to just work similar to `pkgx python`, without caring that some past versions are packaged differently.

This is just my UX/DX feedback though. I understand that it might be hard to do this differently with the current tech. The concrete issue here is that the error message is unhelpful.

**Comment by jhheider** [comment]
No, you're not wrong that it's somewhat related to the first idea. pkgx is absolutely the second style; we've just taken the position (supported by the separate GitHub repos), that those two yarn projects are different projects.

You've hit one of a very small number of case (the three mentioned I believe) that suffer from that. The error message is trying to help you, but it's a complicated answer.

**Comment by mxcl** [comment]
It's not confusing design, it's confusing DX. The design is very good, but the end user result is confusing. We can do better. But not right now. Yarn is the only package this happens to and we did it wrong.

**Comment by JanPokorny** [comment]
@mxcl You have misunderstood. The issue is primarily about _wrong command being show in the hint_, not how you handle Yarn.

![Image](https://github.com/user-attachments/assets/9b120aed-9104-49a7-91f8-3b3b8edef59f)

Regardless of how you handle Yarn, these two suggested commands are wrong and lead to the same error. The CLI should not suggest wrong commands -- that's the main issue here.

**Comment by mxcl** [comment]
pkgx^2 does not have this issue.

**Comment by JanPokorny** [comment]
@mxcl Yep, you're right. I did not `brew upgrade` in a while.

I wouldn't call this a win, though. The error message now is very bare and indecipherable to casual users. I suppose this is subject to change?

![Image](https://github.com/user-attachments/assets/12c8f165-b2a5-4c18-9924-e166e41609db)

**Comment by mxcl** [comment]
yeah I'll be working on error messages in the next few days
